code/datums/components/storage/storage.dm

```diff
@@ -231,9 +231,12 @@
 	progress.update(progress.goal - things.len)
 	return FALSE
 
+/mob/proc/canUseStorage(var/atom/A)
+	return (A.Adjacent(src) && A.loc != src && ishuman(src) && stat == CONSCIOUS && !restrained() && canmove)

```

Comment:
Not exactly what i had in mind, the common checks should stay in component code, here should be just TRUE/FALSE or conditions depending on mob properties like having hands and such for simple animals.  Then you need to add overrides for these for camera mobs that just return FALSE

Reply:
oooo
ok

Reply:
Should I keep checks like restrained() or stat != CONSCIOUS in the mob/proc/canUseStorage(), or have it be part of the common check in storage.dm()